docs/testing/end-to-end-tests.md

```diff
@@ -0,0 +1,31 @@
+# Overview
+
+<!-- TOC -->
+
+- [Overview](#overview)
+    - [What is this?](#what-is-this)
+    - [Our Goals](#our-goals)
+    - [What is tested?](#what-is-tested)
+
+<!-- /TOC -->
+
+## What is this?
+
+End-to-end tests (`e2e` for short) describes the automated functional tests that simulate user interaction with the system under test.
+
+A well-maintained suite of e2e tests, in conjunction with CI (continuous integration) pipelines will ensure that regressions to key user flows are caught as early as possible in the development cycle.
+
+## Our Goals
+
+To accelerate development by being a force for continuous improvement, and help identify, prioritize and mitigate bottlenecks from the system.
+
+<sup>(taken from Quality Squad internal memo)</sup>
+
+## What is tested?
+
+These end-to-end test focus on several key areas:
+
+- Calypso features.
+- interaction between Calypso and Gutenberg editor.
+- internationalization and localization.
+- Gutenberg blocks added by WordPress.com.
```

Comment:
Out of curiosity, what's the purpose of this specific doc? It doesn't seem like it's linked to from anywhere. Could we just add any of these bits to the overview doc in `/test/e2e`?

Reply:
Hmm, I don't recall anymore, it may be an artifact from when I previously looked at moving the entire `test/e2e` directory.

Reply:
Removed.